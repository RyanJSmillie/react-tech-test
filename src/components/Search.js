// eslint-disable-next-line no-unused-vars
import React, { useEffect, useState } from "react";

function Search() {
  return (
    <label htmlFor="search">
      <input className="searchInput" type="text" placeholder="Search" />
    </label>
  );
}

export default Search;
