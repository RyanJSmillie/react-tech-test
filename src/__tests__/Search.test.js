import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";
import setSearchResults from "../components/App";

describe("Search", () => {
  const { asFragment } = render(<Search setSearchResults={setSearchResults} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("button displays correct text", () => {
    const { getByText } = render(
      <Search setSearchResults={setSearchResults} />
    );

    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
