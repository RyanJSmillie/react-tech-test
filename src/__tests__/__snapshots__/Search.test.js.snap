// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <label
    for="search"
  >
    <input
      class="searchInput"
      placeholder="Search"
      type="text"
    />
  </label>
</DocumentFragment>
`;
